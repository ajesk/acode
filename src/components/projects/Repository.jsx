import React from 'react';
import './Repository.scss';

const Repository = ({ repo }) => {
  return (
    <div className="repo" key={repo.id}>
      <a className="repo-link" href={repo.html_url}>{repo.name}</a>
      <div className="repo-details">
        <div className="repo-left">
          <div>{repo.language ? 'Language:' : ''}</div>
          <div>Last Updated:</div>
        </div>
        <div className="repo-right">
          <div>{repo.language}</div>
          <div>{repo.updated_at}</div>
        </div>
      </div>
    </div>
  );
}

export default Repository;
