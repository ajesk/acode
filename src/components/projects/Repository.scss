.repo {
  display: flex;
  flex-direction: column;

  flex: 1 0 auto;

  border: 1px solid #ccc;
  border-radius: 16px;
  padding: 0.5em;
  margin: 0.25em;
}

.repo-details {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.repo-left, .repo-right {
  display: flex;
  flex-direction: column;
  padding: 0.25em;
}