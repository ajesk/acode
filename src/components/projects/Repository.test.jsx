import React from 'react';
import { render } from '@testing-library/react';
import Repository from './Repository';

describe('repository', () => {
  const repo = {
    id: 'ajdflkah392iauk',
    html_url: 'fackall',
    name: 'adsfn,vxmz,mnlknf dsfsdf',
    language: 'today',
    updated_at: 'today'
  };
  test('snapshot', () => {
    expect(render(<Repository repo={repo} />)).toMatchSnapshot();
  });
});