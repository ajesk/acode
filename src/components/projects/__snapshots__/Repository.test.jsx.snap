// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repository snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="repo"
      >
        <a
          href="fackall"
        >
          adsfn,vxmz,mnlknf dsfsdf
        </a>
        <div
          class="repo-details"
        >
          <div
            class="repo-left"
          >
            <div>
              Language:
            </div>
            <div>
              Last Updated:
            </div>
          </div>
          <div
            class="repo-right"
          >
            <div>
              today
            </div>
            <div>
              today
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="repo"
    >
      <a
        href="fackall"
      >
        adsfn,vxmz,mnlknf dsfsdf
      </a>
      <div
        class="repo-details"
      >
        <div
          class="repo-left"
        >
          <div>
            Language:
          </div>
          <div>
            Last Updated:
          </div>
        </div>
        <div
          class="repo-right"
        >
          <div>
            today
          </div>
          <div>
            today
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
